resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -387,7 +446,7 @@ private[spark] class Client(
         .withLabels(driverKubernetesSelectors)
         .endMetadata()
       .withNewSpec()
-        .withType("NodePort")
+        .withType(if (useIngress) "ClusterIP" else "NodePort")
```

Comment:
This works with some but not all ingress. See https://github.com/kubernetes/ingress/blob/master/docs/faq/gce.md#i-created-an-ingress-and-nothing-happens-now-what, https://github.com/kubernetes/kubernetes/issues/26508

I think we'd want the user to get a cluster-ip type here, with the default still being NodePort.